test/fixtures/read-stream.js

```diff
@@ -0,0 +1,37 @@
+
+/**
```

Comment:
This applies to all tests. Is there a mechanism, that will fail the test after a certain time, if it is hanging. I do not spot one at first glimpse. But this is crucial, in order to exit the CI/CD pipeline early and not waste resources and also to give the developer immediate feedback. The old tests have this mechanism. (And such problems occur very often during development on some platforms...). Also a check of the runtime, that it does not exceed a certain value would be good, to spot performance problems with the two eventloops.

Reply:
Mocha has built in timeouts for all tests, the default is 2s.

It can be overridden per suite with:

```js
describe('a module', function () {
  this.timeout(5000) // all tests in the suite will now timeout after 5s

  it('should do a thing', () => {
    // ...
  })
})
```

Or per test with:

```js
describe('a module', () => {
  
  it('should do a thing', function () {
    this.timeout(5000) // only this test will timeout after 5s, all others use the default of 2s
    // ...
  })
})
```

Reply:
Perfect, because these timeouts are really important for the test. It is really often, that something blocks.